modules/fluid_properties/src/userobjects/NaKFluidProperties.C

```diff
@@ -0,0 +1,270 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NaKFluidProperties.h"
+
+registerMooseObject("FluidPropertiesApp", NaKFluidProperties);
+
+InputParameters
+NaKFluidProperties::validParams()
+{
+  InputParameters params = SinglePhaseFluidProperties::validParams();
+  params.addClassDescription("Fluid properties for NaK");
+  params.addRequiredParam<Real>(
+      "weight_fraction_K",
+      "Weight fraction of potassium in NaK. Only eutectic is implemented (X_K = 0.778)");
+  return params;
+}
+
+NaKFluidProperties::NaKFluidProperties(const InputParameters & parameters)
+  : SinglePhaseFluidProperties(parameters), _MNaK((39.102 + 22.9898) / 1000)
+{
+  // We did not implement other weight fractions
+  const Real Xk = getParam<Real>("weight_fraction_K");
+  if (Xk != 0.778)
+    paramError("weight_fraction_K",
+               "Only 0.778 weight percent potassium (eutectic) is implemented");
+
+  // Compute mole fractions from mass fractions
+  const Real AwK = 39.102;
+  const Real AwNa = 22.9898;
+  _Nk = Xk / AwK / (Xk / AwK + (1 - Xk) / AwNa);
+}
+
+NaKFluidProperties::~NaKFluidProperties() {}
+
+std::string
+NaKFluidProperties::fluidName() const
+{
+  return "NaK";
+}
+
+Real
+NaKFluidProperties::molarMass() const
+{
+  return _MNaK;
+}
+
+Real
+NaKFluidProperties::T_from_p_h(Real pressure, Real enthalpy) const
+{
+  auto lambda = [&](Real p, Real current_T, Real & new_h, Real & dh_dp, Real & dh_dT)
+  { h_from_p_T(p, current_T, new_h, dh_dp, dh_dT); };
+  Real T = FluidPropertiesUtils::NewtonSolve(
+               pressure, enthalpy, _T_initial_guess, _tolerance, lambda, name() + "::T_from_p_h")
+               .first;
```

Comment:
It makes me feel better to see this getting used

Reply:
We could avoid it here. It s a third order polynomial there's a method to solve them.